### PR TITLE
chore: upgrade `builder` to `go/v4`

### PR DESCRIPTION
**Pull request by giautm**
https://book.kubebuilder.io/migration/manually_migration_guide_gov3_to_gov4

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -63,31 +63,6 @@ rules:
   - db.atlasgo.io
   resources:
   - atlasmigrations
-  verbs:
-  - create
```
Comment (rotemtam) [inline config/rbac/role.yaml:67]: ?

Comment (giautm) [reply config/rbac/role.yaml:67]: Yep, checking again why it get remove

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -100,12 +75,14 @@ rules:
...
   - db.atlasgo.io
   resources:
+  - atlasmigrations/finalizers
   - atlasschemas/finalizers
   verbs:
   - update
 - apiGroups:
   - db.atlasgo.io
   resources:
+  - atlasmigrations/status
```
Comment (giautm) [inline config/rbac/role.yaml:85]: @rotemtam it's grouped into here.

**Review by datdao — APPROVED** [review]
LGTM

**Pull request merged**